services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -408,28 +408,44 @@ const deleteIndexedBlocks = async job => {
 	try {
 		await BluebirdPromise.map(
 			blocks,
-			async block => {
-				const blockFromNode = await getBlockByHeight(block.height);
-				if (blockFromNode.id === block.id) return;
-
-				// Check if deleted block is indexed and reschedule job if not deleted block is not indexed
-				const [deletedBlockFromDB] = await blocksTable.find({ height: block.height, limit: 1 });
-				if (!deletedBlockFromDB) {
-					// eslint-disable-next-line no-use-before-define
-					await scheduleBlockDeletion(block);
+			async blockFromJob => {
+				const blockFromNode = await getBlockByHeight(blockFromJob.height);
+
+				// Check if deleted block is indexed
+				const [blockFromDB] = await blocksTable.find({ height: blockFromJob.height, limit: 1 });
```

Comment:
```suggestion
				const [blockFromDB] = await blocksTable.find({ height: blockFromJob.height, limit: 1 }, ['id']);
```

Reply:
we do not have a unique constraint on `id` but have a unique constraint on `height`.